### PR TITLE
Migration of experimentalWarnings and sessionPlayer to ES6 modules

### PR DESCRIPTION
**Pull request by Camc314**
**Issues**
part of #868

### CODE REVIEW

**Comment by Camc314** [comment]
Lint is fixed in #1675

File: src/plugins/sessionPlayer/plugin.js [hunk 1]
```diff
@@ -276,251 +274,230 @@ define(['playbackManager', 'events', 'serverNotifications', 'connectionManager']
...
-
-    SessionPlayer.prototype.getRepeatMode = function () {
+    }
 
-    };
+    setMaxStreamingBitrate(options) {
+    }
 
-    SessionPlayer.prototype.setRepeatMode = function (mode) {
+    setRepeatMode(mode) {
```
Comment (dmitrylyzo) [inline src/plugins/sessionPlayer/plugin.js:450]: Has `set`, but not `get` :confused: 
Is this method called for non-local player?
Or is it here just to be safe?

File: src/plugins/sessionPlayer/plugin.js [hunk 2]
```diff
@@ -276,251 +274,230 @@ define(['playbackManager', 'events', 'serverNotifications', 'connectionManager']
...
-
-    SessionPlayer.prototype.setMaxStreamingBitrate = function (options) {
-
-    };
-
-    SessionPlayer.prototype.isFullscreen = function () {
-
-    };
-
-    SessionPlayer.prototype.toggleFullscreen = function () {
```
[the remark this answers is not in the document]
Comment (Camc314) [reply src/plugins/sessionPlayer/plugin.js:461]: Fixed

File: src/plugins/sessionPlayer/plugin.js [hunk 3]
```diff
@@ -276,251 +274,230 @@ define(['playbackManager', 'events', 'serverNotifications', 'connectionManager']
...
-
-    };
-
-    SessionPlayer.prototype.getRepeatMode = function () {
+    }
 
-    };
+    setMaxStreamingBitrate(options) {
+    }
```
[the remark this answers is not in the document]
Comment (Camc314) [reply src/plugins/sessionPlayer/plugin.js]: Nope, it's not needed, I'll remove it.

File: src/plugins/sessionPlayer/plugin.js (hunk 1 above)
Comment (Camc314) [reply src/plugins/sessionPlayer/plugin.js:447]: I'm not too sure, I left it in just to be safe. If it isn't used, it could be removed.

Comment (dmitrylyzo) [reply src/plugins/sessionPlayer/plugin.js:447]: They (get/set) are used with same conditions.
https://github.com/jellyfin/jellyfin-web/blob/1339bb97f4161cddf2c9057a613c99f278e07bfc/src/components/playback/playbackmanager.js#L3552-L3567

`enableLocalPlaylistManagement` will return `false` for `sessionPlayer`.
https://github.com/jellyfin/jellyfin-web/blob/1339bb97f4161cddf2c9057a613c99f278e07bfc/src/components/playback/playbackmanager.js#L14-L24

So both are required.

File: src/plugins/sessionPlayer/plugin.js [hunk 4]
```diff
@@ -276,251 +274,227 @@ define(['playbackManager', 'events', 'serverNotifications', 'connectionManager']
...
-    };
+    }
 
-    SessionPlayer.prototype.getPlaylist = function () {
+    getPlaylist() {
         return Promise.resolve([]);
-    };
-
-    SessionPlayer.prototype.getCurrentPlaylistItemId = function () {
-    };
```
Comment (dmitrylyzo) [inline src/plugins/sessionPlayer/plugin.js:511]: It seems this one is required too.
https://github.com/jellyfin/jellyfin-web/blob/1339bb97f4161cddf2c9057a613c99f278e07bfc/src/components/playback/playbackmanager.js#L2562-L2564

File: src/plugins/sessionPlayer/plugin.js (hunk 1 above)
Comment (Camc314) [reply src/plugins/sessionPlayer/plugin.js:447]: I have added `getRepeatMode` to sessionPlayer

File: src/plugins/sessionPlayer/plugin.js (hunk 4 above)
Comment (Camc314) [reply src/plugins/sessionPlayer/plugin.js:511]: Added Back

**Pull request merged**